### ISSUE TITLE
Redundant `: Copy` trait bound leads to "cannot move"

### ISSUE THREAD

**Issue by dnbln** [intellij-rust/intellij-rust#8439]
## Environment

* **IntelliJ Rust plugin version:** 0.4.163.4378-213
* **Rust toolchain version:** 1.60.0-nightly (a7e2e3396 2022-01-08)
* **IDE name and version:** CLion Build #CL-213.6461.46, built on December 22, 2021
* **Operating system:** Linux

## Problem description
"Cannot move" in valid code

## Steps to reproduce

Start with https://gist.github.com/dnbln/a76c0639616b2b183a472e757efbc01b#file-bug-rs

Observe the "cannot move" error on [lines 19-20](https://gist.github.com/dnbln/a76c0639616b2b183a472e757efbc01b#file-bug-rs-L19-L20):
![image](https://user-images.githubusercontent.com/39565527/150421431-6362105e-cb8e-4e3c-bbce-e7caeae06818.png)

But if I remove the `where` clause on [lines 14-15](https://gist.github.com/dnbln/a76c0639616b2b183a472e757efbc01b#file-bug-rs-L14-L15), then it goes away.

Both of them compile just fine

**Comment by ortem** [comment]
@dnbln Thank you for the detailed report!

@vlad20012 It looks like our trait resolution always fails in case of redundant trait bound, e.g. in the following case:
```Rust
trait X: Copy {}

struct S<A: X> { a: A }

impl<A: X> S<A>
    where A: Copy // without this line, plugin recognizes `A` as copyable
{
    fn foo(&mut self) {
        self.a; // false-positive "Cannot move"
    }
}
```

**Comment by afetisov** [comment]
Doesn't seem to reproduce on the latest plugin. Close?

**Comment by ortem** [comment]
@afetisov Seems so, thank you!
